pipeline-runner/src/dataIntegration.r

```diff
@@ -185,13 +200,13 @@ run_dataIntegration <- function(scdata, config){
     scdata@misc[["active.reduction"]] <- active.reduction
 
     # Scale in order to compute PCA
-    scdata <- Seurat::ScaleData(scdata, verbose = F)
-
-    # HARDCODE numPCs to 50
-    scdata <- Seurat::RunPCA(scdata, npcs = 50, features = Seurat::VariableFeatures(object=scdata), verbose=FALSE)
+    if (!have.pca) {
+        scdata <- Seurat::ScaleData(scdata, verbose = FALSE)
+        scdata <- Seurat::RunPCA(scdata, verbose=FALSE)
```

Comment:
I would leave here the comment related to the default value of PCs, just to have it in mind.

Reply:
Thanks @JuanluOnieva-biomage I added the mocks for harmony in the api. I'm also going to refactor a bit as the integration function is getting a bit unwieldly.